File: q8/q8.cpp
class Solution {
public:
    int myAtoi(string str) {
        int i;
        bool negative = false;
        for (i = 0; i < str.size(); i++){
            if (str[i] != ' '){
                break;
            }
        }
        if(i == str.size())
            return 0;
        if(str[i] == '-' && i+1 != str.size()){
            negative = true;
            return findnum (str.substr(i+1, str.size()), negative);
        }
        if (str[i] == '+' && i+1 != str.size())
            return findnum (str.substr(i+1, str.size()), negative);
        return findnum (str.substr(i, str.size()), negative);
        
    }
    int findnum (string str, bool negative){
        if (str.size() == 0)
            return 0;
        int num = 0;
        for(int i = 0; i < str.size(); i++){
            if( int(str[i]) < 58 && int(str[i]) > 47 ){
                if(!negative && (num > INT_MAX/10 || (num==INT_MAX/10 && (str[i]-'0')>=7))){
                    return INT_MAX;
                }
                if(negative && (num > INT_MAX/10 || (num==INT_MAX/10 && (str[i]-'0')>=8))){
                    return INT_MIN;
                }
                num = num*10 + (str[i] - '0');
            }
            else
                break;
        }
        if(negative)
            return -num;
        return num;
        
    }
};

File: q8/q8.py
class Solution:
    def myAtoi(self, str: str) -> int:
        str = str.strip()
        if len(str) == 0:
            return 0
        negative = False
        if(str[0] == '-'):
            negative = True
            str = str[1:]
        elif(str[0] == '+'):
            nagative = False
            str = str[1:]
        solution = 0
        for i in range(len(str)):
            if str[i].isdigit():
                solution = int (solution*10 + ord(str[i]) - 48)
            else:
                break
        if negative:
            solution = -solution
        if solution > 2**31 - 1:
            solution = (2**31 - 1)
        elif solution < -1 * 2**31:
            solution = (-1 * 2**31)
        return solution
